tests/assoc_sim.R

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/Rscript
+#
+# Copyright 2015 Google Inc. All rights reserved.
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+#     http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+library("optparse")
+
+# First parse args
+options(stringsAsFactors = FALSE)
+if(!interactive()) {
+  option_list <- list(
+    # Flags
+    make_option(c("--candidates", "-c"), default = "candidates.csv",
+                help = "Filename for candidates"),
+    make_option(c("--params", "-p"), default = "params.csv",
+                help = "Filename for parameters"),
+    make_option(c("--reports", "-r"), default = "reports.csv",
+                help = "Filename for reports"),
+    make_option(c("--map", "-m"), default = "map",
+                help = "Filename *prefix* for map(s)"),
+    make_option(c("--num", "-n"), default = 1e05,
+                help = "Number of reports"),
+    make_option(c("--unif", "-u"), default = FALSE,
+                help = "Run simulation with uniform distribution")
+  )
+  opts <- parse_args(OptionParser(option_list = option_list))
+}    
+
+source("../analysis/R/encode.R")
+source("../analysis/R/decode.R")
+source("../analysis/R/simulation.R")
+source("../analysis/R/read_input.R")
+source("../analysis/R/association.R")
+
+# Read candidates from a csv file
+# Inputs: filename. The file is expected to contain two rows of strings
+#         (one for each variable):
+#         "google.com", "apple.com", ...
+#         "ssl", "nossl", ...
+# Returns: a list containing strings
+GetCandidatesFromFile <- function(filename) {
```

Comment:
No, they shouldn't be called "candidates".  That is the user provides; it's not what we're doing here in the simulation.

What you're talking about is incidental -- if it's the uniform distribution it will always be exact.

Just call them GetUniqueValues and unique_values if you can't think of a better name.